src/index.ts

```diff
@@ -104,17 +140,31 @@ export class SgidClient {
    * @param code The authorization code received from the authorization server
    * @param nonce Specify null if no nonce
    * @param redirectUri The redirect URI used in the authorization request, defaults to the one registered with the client
+   * @param codeVerifier The code verifier that was used to generate the code challenge that was passed in `authorizationUrl`
    * @returns The sub of the user and access token
    */
-  async callback(
-    code: string,
-    nonce: string | null = null,
+  async callback({
+    code,
+    nonce = null,
     redirectUri = this.getFirstRedirectUri(),
-  ): Promise<{ sub: string; accessToken: string }> {
+    codeVerifier,
+  }: CallbackParams): Promise<{ sub: string; accessToken: string }> {
+    if (this.apiVersion !== 2) {
+      throw new Error(
+        "Code verifier must be provided in 'callback' when using apiVersion 2",
+      )
+    }
+
+    if (codeVerifier === undefined) {
+      throw new Error(
+        "Code verifier must be provided in 'callback' when using apiVersion 2",
```

Comment:
Since we're not giving them a choice to choose the api version, I'm not sure if we need to mention it here. what do you think?

Reply:
actually since the `codeVerifier` should no longer be optional with PKCE, do you think we should make it not optional in the type signature and also remove this check?

Reply:
same for the check for `codeChallenge === undefined` above in the `authorizationUrl` function

Reply:
same for the check for `codeChallenge === undefined` above in the `authorizationUrl` function